bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeApi.java

```diff
@@ -0,0 +1,83 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.senechome.internal;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.http.MimeTypes;
+import org.openhab.binding.senechome.internal.json.SenecHomeResponse;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link SenecHomeApi} class configures http client and
+ * performs status requests
+ *
+ * @author Steven.Schwarznau - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class SenecHomeApi {
+    private static final String HTTP_PROTO_PREFIX = "http://";
+
+    private final HttpClient httpClient;
+    private final Gson gson = new Gson();
+    private String hostname = "";
+
+    public SenecHomeApi(HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    public void setHostname(String hostname) {
+        this.hostname = hostname;
+    }
+
+    /**
+     * POST json with empty, but expected fields, to lala.cgi of Senec webinterface
+     * the response will contain the same fields, but with the corresponding values
+     *
+     * To receive new values, just modify the Json objects and add them to the thing channels
+     *
+     * @param hostname Hostname or ip address of senec battery
+     * @return Instance of SenecHomeResponse
+     * @throws MalformedURLException Configuration/URL is wrong
+     * @throws IOException Communication failed
+     */
+    public SenecHomeResponse getStatistics()
+            throws InterruptedException, TimeoutException, ExecutionException, IOException {
+        String location = HTTP_PROTO_PREFIX + hostname;
+
+        Request request = httpClient.newRequest(location);
+        request.header(HttpHeader.ACCEPT, MimeTypes.Type.APPLICATION_JSON.asString());
+        request.header(HttpHeader.CONTENT_TYPE, MimeTypes.Type.FORM_ENCODED.asString());
+        ContentResponse response = request.method(HttpMethod.POST)
+                .content(new StringContentProvider(gson.toJson(new SenecHomeResponse()))).send();
+
+        if (response.getStatus() == HttpStatus.OK_200) {
+            return gson.fromJson(response.getContentAsString(), SenecHomeResponse.class);
+        } else {
+            throw new IOException("Got unexpected response code " + response.getStatus());
```

Comment:
Do you think it would be useful to have the content logged in case of a failure? Maybe at TRACE level

Reply:
Fine for me, changed also another log from warning to trace as most likely no one is interested into that message...